docs/installation/local.md

```diff
@@ -2,169 +2,177 @@
 
 ## Dependencies required to run Orga Server
 
-* Python 3.7
-* Postgres
-* OpenSSL
+-   Python 3.7
+-   Postgres
+-   OpenSSL
 
-### For mac users
-```sh
-brew install postgresql
-````
-
-### For debian-based linux users
-```sh
-sudo apt-get update
-sudo apt-get install postgresql postgresql-contrib libssl-dev
-```
-
-## Steps
+## Get a copy of source code
 
-Make sure you have the dependencies mentioned above installed before proceeding further.
+- Clone the Open Event Server repository (from the development branch) and `cd ` into the directory.
 
-* **Step 0** - Clone the Open Event Server repository (from the development branch) and ```cd ``` into the directory.
 ```sh
-git clone -b development https://github.com/fossasia/open-event-server.git
+git clone https://github.com/fossasia/open-event-server.git
 cd open-event-server
 ```
-**Note :** If you want to contribute, first fork the original repository and clone the forked repository into your local machine followed by ```cd``` into the directory
+
+- If you want to contribute, first fork the original repository to your GitHub profile and clone that fork into your local machine, followed by `cd` into the directory
+
 ```sh
-git clone https://github.com/USERNAME/open-event-server.git
+git clone git@github.com:USERNAME/open-event-server.git
 cd open-event-server
 ```
 
-* **Step 1** - Install python3 requirements. You need to be present in the root directory of the project.
+- Tip:
 
-# System Wide Installation
+  + Setup SSH key in your profile, and use SSH method to clone the source code, so that you don't have to type password repeatly.
 
-```sh
-sudo -H pip3 install -r requirements.txt
-```
-hint: You may need to upgrade your pip version and install following packages if you encounter errors while installing the requirements.
+  + To let your personal fork up-to-date with the original FOSSASIA repo, add original repo to "remote" list and regularly fetch its new content.
 
-**Note:** For Mac OS Sierra users, if you get an error that 'openssl/aes.h' could not be found when installing requirements.txt using pip, try the steps shown here - [OSX openssl header error](https://tutorials.technology/solved_errors/1-OSX-openssl_opensslv_h-file-not-found.html)
+  ```sh
+  git remote add upstream https://github.com/fossasia/open-event-server.git
+  git fetch upstream
+  git merge upstream/development
```

Comment:
```suggestion
  git pull -r upstream development
```

Reply:
Are you sure you want `git pull -r`? Because `-r` means `rebase`, and `rebase` will rewrite Git history, which later on, will cause a lot of messes.

Reply:
It would just rebase their branch with `upstream development` (Which should ideally be the case if they make a new branch to solve an issue, and if the developer is in the `development` branch itself, it would just update it). Why would it cause a mess?

Reply:
Yes, this is important because merge will merge their local changes on development branch with upstream changes and that will mean their branches made from development branch will contain unnecessary changes from other PRs. We should explicitly state that you need to create a new branch for working, not committing on development branch, and this will ensure this doesn't happen. Besides, even if they run the command, the work will not be lost, rather replayed after the upstream commits

Reply:
@prabhupant  I got lost in this discussion thread, so I replied in https://github.com/fossasia/open-event-server/pull/7320#discussion_r502944325

@iamareebjamal Is your comment to answer @prabhupant or me?

 > "contain unnecessary changes from other PRs."

The changes you referred as "unnecessary" are actually the ones which went into "upstream" FOSSASIA repo, it is not "unnecessary" (if it is unnecessary, it wouldn't be accepted to upstream).

If you don't incorporate those changes to your branch, later on, you will still do so, or your PR will be in "conflict" state. The later you merge (from upstream), the bigger that conflict grows, and later you have to make bigger effort to solve the conflict.

On the other hand, those "upstream" change is not "unnecessary" because you have to make sure your code is in harmony with code of other parts. Merging early will let you know early what need to change to make sure your code still work well with other parts.

Reply:
This is when they have created a different branch. Your assumption of people starting contribution is that they understand best practices of git. Most people beginning here make PRs from the development branch. If they run fetch merge (which is basically pull), then it'll be like dev-commit -> their-commit -> dev-latest-commit -> merge-commit

Which we don't want. Now their local development branch contains interleaved unnecessary commits. 

Also, we want people to rebase their local branch with upstream branch across works, not merge. Merges are disastrous with database migrations. We want their commits to look like:

dev1 - dev2 - dev3 - my1 - my2 - my3

With merge, it'll look like

dev1 - my1 - m2 - dev2 - my3 - dev3

Which we don't want and can also cause wrong migration sequences. It also makes it difficult to squash commits as their commits are not grouped together. And checking out a commit will give them wrong state of the project. Checking out on my3 will include changes from dev2 but not dev3 which is already merged in the backend

Reply:
Ok. Because you guys are the ones who review the PR, I should follow the approach that you are familiar with.